### ISSUE TITLE
--pause-after-load throws, execution halted, tests cannot be ran

### ISSUE THREAD

**Issue by daniel-v** [dart-lang/test#362]
Command: `pub run test:test --pause-after-load --pub-serve=8083 -p dartium`
Exception:
```
ailed to run Dartium: 'package:test/src/runner/browser/dartium.dart': malformed type: line 148: type 'CancelableCompleter' is not loaded
      var completer = new CancelableCompleter(() {
                          ^
  type error..
  dart:isolate  _RawReceivePortImpl._handleMessage
  
  Bad state: Future already completed
  dart:isolate  _RawReceivePortImpl._handleMessage
  
  Bad state: Future already completed
  dart:isolate  _RawReceivePortImpl._handleMessage
  
  Bad state: Future already completed
  dart:isolate  _RawReceivePortImpl._handleMessage
```

**Dart version:** `0.12.2`
**OS:** `Linux  3.16.0-38-generic #52~14.04.1-Ubuntu SMP Fri May 8 09:43:57 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`
**Test package version:** `0.12.5+1`

I dag into it: the cause of the issue is that there are duplicate declarations of certain classes, such as `CancelableCompleter`, `CancelableFuture` in the test package (in util). The same classes, with slightly different signitures are being imported from the `async` package as well.

These modifications seem to fix the issue: https://gist.github.com/daniel-v/476adb6839944cc5333d

I signed the CLA. (If I knew how to properly contribute this, I would ;) )

**Comment by nex3** [comment]
Thanks for the patch! It turns out this was already fixed by e4f82b38523feaeb9be5e7b6f1dd77bc91624d12 and released as 0.12.5+2. You should be able to get the fix by just upgrading.

**Comment by daniel-v** [comment]
Hey Natalie,
thanks a bunch!
